Sprint-2/stretch/count-words.js

```diff
@@ -26,3 +26,45 @@
 
 3. Order the results to find out which word is the most common in the input
 */
+
+
+
+// Function clean and normalize the input string (remove punctuations and convert to all lower case)
+function removePunctuations(inputStr) {
+  return inputStr
+    .replace(/[.,!?]/g, '')
+    .toLowerCase();
+}
+// Function to split a string into words
+function splitToWords(removePunctuations) {
+  return removePunctuations.split(/\s+/);
+}
+
+// Function to count occurences of words in an array
+// Function to count occurrences of words in an array
+function countWordNumber(words) {
+  const wordCount = {};
+
+  words.forEach(word => {
+    if (wordCount[word]) {
+      wordCount[word] += 1;
+    } else {
+      wordCount[word] = 1;
+    }
+  });
+  return wordCount;
+}
+// Function to sort word counts by frequency
+function sortWords(wordCount) {
+  return Object.fromEntries(
+    Object.entries(wordCount).sort((a, b) => b[1] - a[1])
```

Comment:
Instead of using `b[1]` and `a[1]`, perhaps you can give them more meaningful name? 
(Array destructuring can also help).